cmd/juju/application/deployer/charm.go

```diff
@@ -364,14 +364,15 @@ func (c *repositoryCharm) PrepareAndDeploy(ctx *cmd.Context, deployAPI DeployerA
 		if c.applicationName != "" {
 			appName = c.applicationName
 		}
+
 		configYAML, err := utils.CombinedConfig(ctx, c.model.Filesystem(), c.configOptions, appName)
 		if err != nil {
 			return errors.Trace(err)
 		}
 
 		info, localPendingResources, errs := deployAPI.DeployFromRepository(application.DeployFromRepositoryArg{
 			CharmName:        c.userRequestedURL.Name,
-			ApplicationName:  c.applicationName,
+			ApplicationName:  appName,
```

Comment:
This is causing `jameinel-ubuntu-lite` charm should deploy as an `jameinel-ubuntu-lite` application. The ApplicationName in the DeployFromRepositoryArgs should always be the user provided only.